File: javarecipe/src/main/java/work/shion/javarecipe/pages/entrypoint/presenters/TutorialPresenter.java
package work.shion.javarecipe.pages.entrypoint.presenters;

import com.annimon.stream.Optional;

import java.lang.ref.WeakReference;

import work.shion.javarecipe.pages.entrypoint.contracts.TutorialPresenterContract;
import work.shion.javarecipe.pages.entrypoint.contracts.TutorialViewContract;


/**
 * チュートリアルの挙動実装
 */
public class TutorialPresenter implements TutorialPresenterContract {

    private final WeakReference<TutorialViewContract> viewer;


    public TutorialPresenter(
            WeakReference<TutorialViewContract> viewer
    ) {
        this.viewer = viewer;
    }


    /**
     * TabPage の呼び出し
     */
    @Override
    public void callTabPage() {
        Optional.ofNullable(viewer.get())
                .ifPresent(viewer -> viewer.goTabPage());
    }
}

File: ktrecipe/src/main/java/work/shion/ktrecipe/pages/entrypoint/presenters/TutorialPresenter.kt
package work.shion.ktrecipe.pages.entrypoint.presenters

import work.shion.ktrecipe.pages.entrypoint.contracts.TutorialPresenterContract
import work.shion.ktrecipe.pages.entrypoint.contracts.TutorialViewContract
import java.lang.ref.WeakReference


/**
 * チュートリアルの挙動実装
 */
class TutorialPresenter(
        private val viewer: WeakReference<TutorialViewContract>
) : TutorialPresenterContract {

    /**
     * TabPage の呼び出し
     */
    override fun callTabPage() {
        viewer.get()?.apply {
            goTabPage()
        }
    }
}
